build-logic/module-identity/src/main/kotlin/gradlebuild/identity/provider/BuildTimestampValueSource.kt

```diff
@@ -43,13 +39,6 @@ abstract class BuildTimestampValueSource : ValueSource<String, BuildTimestampVal
     }
 
     override fun obtain(): String? = parameters.run {
-
-        val buildTimestampFromReceipt = buildTimestampFromBuildReceipt.orNull
```

Comment:
We might run into an issue here now that we're not consuming _some_ build timestamp from prior builds in CI. We had a prior problem (see [fix](https://github.com/gradle/gradle/pull/19273/files)) where during local docs testing we reached the default case below where we return `Date().withoutTime()`. For the cases where we previously consumed the build timestamp from the receipt, can we be sure that we don't reach that default case? (This would be during github workflows and the team city CI builds). I see below we're checking `runningOnCi`, but I'm not sure if that is set when running the github workflow builds. 

Though, this sparks the question on whether we need to still consume the prior timestamps in subsequent builds. @gradle/bt-developer-productivity Do you know why we are keeping track of the initial build timestamp and then subsequently passing that to later builds in a chain? What is the significance of later builds knowing the timestamp of the initial build?

Reply:
There are two use-cases for the build timestamp which we pass from build to build:
- Determine the version number from this timestamp in the case of a nightly <- This needs to stay, so the version number of the distribution we test doesn't change.
- Pack the build timestamp into the distribution <- This is what we can remove, since the distribution doesn't need to know about the timestamp any more when we don't print it out.

Reply:
So realistically that means for this PR we probably still want to keep the code which loads timestamps from build to build, but instead does so through a "build timestamp file" that doesn't eventually get embedded into the distribution like the receipt does.

Reply:
That is what already happens right now I think: it uses a build receipt not embedded in the build to load the build timestamp from in the `incoming` directory.